src/Stackify/Log/Entities/Api/WebRequestDetail.php

```diff
@@ -93,19 +130,71 @@ private function __construct()
         $this->RequestUrl = $this->getRequestUrl();
         $this->RequestUrlRoot = filter_input(INPUT_SERVER, 'SERVER_NAME');
         $this->ReferralUrl = filter_input(INPUT_SERVER, 'HTTP_REFERER');
-        $this->Headers = $this->getHeaders();
-        $this->Cookies = isset($_COOKIE) ? self::getRequestMap($_COOKIE, true) : null;
-        $this->QueryString = isset($_GET) ? self::getRequestMap($_GET) : null;
-        $this->PostData = isset($_POST) ? self::getRequestMap($_POST) : null;
-        $this->SessionData = isset($_SESSION) ? self::getRequestMap($_SESSION, true) : null;
-        $this->PostDataRaw = file_get_contents('php://input');
+
+        /**
+         * @var \Stackify\Log\Transport\Config\Agent
+         */
+        $agentConfig = Agent::getInstance();
+
+        if ($agentConfig) {
+            $this->Headers = $agentConfig->getCaptureErrorHeaders()
+                ? $this->getHeaders(
+                    $agentConfig->getCaptureErrorHeadersBlacklist(),
+                    $agentConfig->getCaptureErrorHeadersWhitelist()
+                )
+                : null;
+            $this->Cookies = isset($_COOKIE) && $agentConfig->getCaptureErrorCookies()
+                ? self::getRequestMap(
+                    $_COOKIE,
+                    $agentConfig->getCaptureErrorCookiesBlacklist(),
+                    $agentConfig->getCaptureErrorCookiesWhitelist()
+                )
+                : null;
+
+            $this->QueryString = isset($_GET) && $agentConfig->getCaptureGetVariables()
+                ? self::getRequestMap(
+                    $_GET,
+                    $agentConfig->getCaptureGetVariablesBlacklist(),
+                    $agentConfig->getCaptureGetVariablesWhitelist()
+                )
+                : null;
+            $this->PostData = isset($_POST) && $agentConfig->getCapturePostVariables()
+                ? self::getRequestMap(
+                    $_POST,
+                    $agentConfig->getCapturePostVariablesBlacklist(),
+                    $agentConfig->getCapturePostVariablesWhitelist()
+                )
+                : null;
+            $this->SessionData = isset($_SESSION) && $agentConfig->getCaptureSessionVariables()
+                ? self::getRequestMap(
+                    $_SESSION,
+                    $agentConfig->getCaptureSessionVariablesBlacklist(),
+                    $agentConfig->getCaptureSessionVariablesWhitelist()
+                )
+                : null;
```

Comment:
These statements are redundant and may cause confusion if they need to be changed in the future. It might be safer to iterate through the magic variables and protected variables. Try putting the getRequestMap call in a closure instead of repeatedly writing the Blacklist & whitelist getters.

Reply:
A better alternative would be to make a list of tuples of the $this property, magic variable, and getCapture* method.
Evaluate each row and assign the result to $this via array index.